core/src/main/java/org/wso2/testgrid/core/util/JsonPropFileUtil.java

```diff
@@ -0,0 +1,584 @@
+     *
+     * NOTE :: JSON File structure
+     * { general: {
+     *      gen_prop1:val1 , gen_prop2:val2
+     *   },
+     *   currentscript: {
+     *      prop4:val4
+     *   },
+     *   script1: {prop3:val3},
+     *   script2: {prop4:val4}
+     *  }
+     *
+     * @param properties properties to be added
+     * @param propFilePath path of the property file
+     * @param jsonFilePath path to the JSON file
+     * @param scriptName  OPTIONAL Name of script file if not provided property added as a general variable
+     * @throws TestPlanExecutorException if writing to the property file fails
+     */
+    public void persistAdditionalInputs(Map properties, Path propFilePath , Path jsonFilePath,
+                                         Optional<String> scriptName) throws TestPlanExecutorException {
+
+        if (!scriptName.isPresent()) {
+            // If value is not specified from a script add the value as a general value to the json file
+            try {
+
+                // If the JSON file exists read existing values and append to the file
+                InputStream jsonInputStream = new FileInputStream(jsonFilePath.toString());
+                JSONTokener jsonTokener = new JSONTokener(jsonInputStream);
+                JSONObject inputJson = new JSONObject(jsonTokener);
+                Iterator it = properties.entrySet().iterator();
+
+                // Add new value to a flatlist level of the script
+                while (it.hasNext()) {
+                    Map.Entry pair = (Map.Entry) it.next();
+                    if (inputJson.has("general")) {
+                        inputJson.getJSONObject("general").put((String) pair.getKey(), pair.getValue());
+                    } else {
+                        JSONObject generalProps = new JSONObject(properties);
+                        inputJson.put("general", generalProps);
+                    }
+                }
+                // Append to json file
+                try (BufferedWriter jsonWriter = Files.newBufferedWriter(Paths.get(jsonFilePath.toString()))) {
+                    inputJson.write(jsonWriter);
+                    jsonWriter.write("\n");
+                } catch (Throwable e) {
+                    logger.error("Error while persisting infra input params to " + jsonFilePath, e);
+                }
+
+            } catch (IOException e) {
+                // If file does not exist create new file and write into it
+                logger.info(jsonFilePath + "created");
+                HashMap<String, Object> newJsonFileInputs = new HashMap<String, Object>();
+                newJsonFileInputs.put("general", properties);
+                JSONObject newJsonFile = new JSONObject(newJsonFileInputs);
+                try (BufferedWriter jsonWriter = Files.newBufferedWriter(Paths.get(jsonFilePath.toString()))) {
+                    newJsonFile.write(jsonWriter);
+                    jsonWriter.write("\n");
+                } catch (Throwable ex) {
```

Comment:
We almost never catch Throwable. Even if we do, then we should add proper justification.
Same goes for Exception as well. We should rather catch specific exceptions.